Adding Enforcement Attributes to Demand Data (#179)
openmobilityfoundation/curb-data-specification: events/README.md

| `event_session_id` | [UUID][uuid] | Optional | May be provided to tie known connected `park_start` and `park_end` event types together by a unique session ID. If _not_ confident of being able to determine a `park_end` event at some time after `park_start` is recorded (i.e., you cannot detect when a vehicle departs), then do _not_ use session_id. This field may be most useful to payment companies who provide their source data as sessions (typical for transaction data). _Note also_: the use of the term "session" across CDS means the start and end of curb usage of a vehicle, not necessarily a financial or payment session or transaction. |

> | `curb_zone_id` | [UUID][uuid] | Conditionally Required | Unique ID of the Curb Zone where the event occurred. Required for events that occurred at a known Curb Zone for ALL _event_types_. |
> | `curb_area_ids` | Array of [UUID][uuid] | Conditionally Required | Unique IDs of the Curb Area where the event occurred. Since Curb Areas can overlap, an event may happen in more than one. Required for events that occurred in a known Curb Area, if known and used, for these event_types: _enter_area, exit_area, park_start, park_end_ |

Reviewer: ```suggestion
| `curb_zone_id` | [UUID][uuid] | Conditionally Required | Unique ID of the Curb Zone where the event occurred. Required for events that occurred in a known Curb Zone, if known and used, for ALL _event_types_. |
| `curb_area_ids` | Array of [UUID][uuid] | Conditionally Required | Unique IDs of the Curb Area where the event occurred. Since Curb Areas can overlap, an event may happen in more than one. Required for events that occurred in a known Curb Area, if known and used, for ALL _event_types_. |
```

Reviewer: Similar to below, I don't believe these are breaking changes since we're saying that if we know and use the space/object ID for any event, it would add more data that can be used/consumed.

> | `curb_space_id` | [UUID][uuid] | Conditionally Required | Unique ID of the Curb Space where the event occurred. Required for events that occurred at a known Curb Space, if known and used, for these event_types: _park_start, park_end, enter_area, exit_area, vehicle_detected, vehicle_violation, citation_given_ |
> | `curb_object_id` | [UUID][uuid] | Conditionally Required | Unique ID of the Curb Object where the event occurred. Required for events that occurred at a known Curb Object, if known and used, for these event_types: _park_start, park_end, enter_area, exit_area, vehicle_detected, vehicle_violation, citation_given_ |

Reviewer: These changes are to align with proposed changes down below where we define the new event types.

```suggestion
| `curb_space_id` | [UUID][uuid] | Conditionally Required | Unique ID of the Curb Space where the event occurred. Required for events that occurred at a known Curb Space, if known and used, for ALL _event_types_. |
| `curb_object_id` | [UUID][uuid] | Conditionally Required | Unique ID of the Curb Object where the event occurred. Required for events that occurred at a known Curb Object, if known and used, for ALL _event_types_. |
```

Reviewer: I don't believe these are breaking changes since we're saying that if we know and use the space/object ID for any event, it would add more data that can be used/consumed. Current implementation conditionally limits for certain event types and I'm not sure there's much value.

| `curb_occupants` | Array of [Curb Occupant](#curb-occupants) | Conditionally Required | Current occupants of the Curb Zone. If the sensor is capable of identifying the linear location of the vehicle, then elements are sorted in ascending order according to the start property of the linear reference. Otherwise, elements appear in no particular order. Required for sources capable of determining it for the following event_types: _park_start, park_end, scheduled_report_ |
| `actual_cost` | Integer | Optional | If available from the source, the actual cost, in the currency defined in currency, paid by the curb user for this event. The currency type is sent in with the [REST Endpoints](#rest-endpoints) JSON object. All costs should be given as integers in the currency's smallest unit. As an example, to represent $1 USD, specify an amount of 100 (for 100 cents). |

> | `enforcement` | Array of [Enforcement][enforcement] objects | Optional | Enforcement information related to this Curb Event, relevant to the moment in time the event happens. Only used for enforcement related events such as `vehicle_detected`, `vehicle_violation`, `citation_given`, and `violation_end`. |

Reviewer: These changes are to align with proposed changes down below where we define the new event types.

```suggestion
| `enforcement` | Array of [Enforcement][enforcement] objects | Optional | Enforcement information related to this Curb Event, relevant to the moment in time the event happens. Only used for enforcement related events such as `vehicle_detected`, `vehicle_violation_start`, `vehicle_violation_end`, and `citation_issued`. |
```

| `scheduled_report` | event source reported status at a scheduled interval |
| `enter_area`       | vehicle enters the relevant geographic area |
| `exit_area`        | vehicle exits the relevant geographic area |

> | `vehicle_detected` | the event of detecting or locating a vehicle at the curb. For example, from a manual surveying or license plate recognition machine |
> | `vehicle_violation` | the event of detecting a vehicle that is doing something illegal or not allowed at the curb. For example, a non permitted vehicle is parked or someone overstayed the time limit. This does not reflect the actual ticket given. Typically requires a `vehicle_detected` event first  |
> | `citation_given`   | the event of providing a ticket or citation to a vehicle. Typically requires a `vehicle_violation` event first |
> | `violation_end`    | the end of a violation, when it is no longer an issue | 

Reviewer 2: The event types `vehicle_violation`, `violation_end`, and `citation_given` have been renamed to `vehicle_violation_start`, `vehicle_violation_end`, and `citation_issued` respectively. These changes:

1. Clarify that the violation events relate specifically to vehicle compliance violations.
2. Improve consistency and precision in naming conventions.
3. Replace “given” with “issued” to reflect the formal creation of a citation, regardless of whether it has been physically delivered or received.
4. Specify that `vehicle_violation_end` applies to time-based violations where an end can be programmatically detected.

```suggestion
| `vehicle_detected` | detection or observation of a vehicle within a curb area. Can originate from manual surveying, automated license plate recognition (LPR) systems, or other detection methods. |
| `vehicle_violation_start` | start of a compliance violation at a curb location, triggered when a vehicle is not permitted or has exceeded allowed time limits. This event may be published after detecting a vehicle but does not require a `vehicle_detected` event. |
| `vehicle_violation_end` | resolution of a compliance violation, when the vehicle is no longer in violation of regulations. Used for time-based violations where the end of the violation can be detected. Requires a preceding vehicle_violation_start event for the same vehicle. | 
| `citation_issued` | issuance of a ticket or citation to a vehicle. May be published in addition to a related vehicle_violation_start event. |
```

> | `vehicle_detected` | the event of detecting or locating a vehicle at the curb. For example, from a manual surveying or license plate recognition machine |
> | `vehicle_violation` | the event of detecting a vehicle that is doing something illegal or not allowed at the curb. For example, a non permitted vehicle is parked or someone overstayed the time limit. This does not reflect the actual ticket given. Typically requires a `vehicle_detected` event first  |
> | `citation_given`   | the event of providing a ticket or citation to a vehicle. Typically requires a `vehicle_violation` event first |
> | `violation_end`    | the end of a violation, when it is no longer an issue | 

Reviewer: The event types `vehicle_violation`, `violation_end`, and `citation_given` have been renamed to `vehicle_violation_start`, `vehicle_violation_end`, and `citation_issued` respectively. These changes:

1. Clarify that the violation events relate specifically to vehicle compliance violations.
2. Improve consistency and precision in naming conventions.
3. Replace “given” with “issued” to reflect the formal creation of a citation, regardless of whether it has been physically delivered or received.
4. Specify that `vehicle_violation_end` applies to time-based violations where an end can be programmatically detected.

```suggestion
| `vehicle_detected` | detection or observation of a vehicle within a curb area. Can originate from manual surveying, automated license plate recognition (LPR) systems, or other detection methods. |
| `vehicle_violation_start` | start of a compliance violation at a curb location, triggered when a vehicle is not permitted or has exceeded allowed time limits. This event may be published after detecting a vehicle but does not require a `vehicle_detected` event. |
| `vehicle_violation_end` | resolution of a compliance violation, when the vehicle is no longer in violation of regulations. Used for time-based violations where the end of the violation can be detected. Requires a preceding vehicle_violation_start event for the same vehicle. | 
| `citation_issued` | issuance of a ticket or citation to a vehicle. May be published in addition to a related vehicle_violation_start event. |
```
